Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -5829,26 +5829,36 @@ static bool isRenderElementWithHDR(const RenderElement& renderer)
             if (cachedImage->hasPaintedHDRContent())
                 return true;
         }
-        return false;
-    }
-
-    if (CheckedPtr imageRenderer = dynamicDowncast<LegacyRenderSVGImage>(renderer)) {
+    } else if (CheckedPtr imageRenderer = dynamicDowncast<LegacyRenderSVGImage>(renderer)) {
         if (auto* cachedImage = imageRenderer->imageResource().cachedImage()) {
             if (cachedImage->hasPaintedHDRContent())
                 return true;
         }
-        return false;
-    }
-
 #if ENABLE(PIXEL_FORMAT_RGBA16F)
-    if (CheckedPtr canavsRenderer = dynamicDowncast<RenderHTMLCanvas>(renderer)) {
+    } else if (CheckedPtr canavsRenderer = dynamicDowncast<RenderHTMLCanvas>(renderer)) {
         if (auto* renderingContext = canavsRenderer->canvasElement().renderingContext()) {
             if (renderingContext->isHDR())
                 return true;
         }
-        return false;
-    }
 #endif
+    }
+
+    CheckedRef style = renderer.style();
+
+    if (style->hasBackgroundImage()) {
+        if (auto* cachedImage = style->backgroundLayers().image()->cachedImage()) {
+            if (cachedImage->hasPaintedHDRContent())
+                return true;
+        }
+    }
+
+    if (style->hasBorderImage()) {
```

Comment:
What about `listStyleImage()`? There may be other images in style I'm forgetting. `RenderElement::styleDidChange()` does some stuff for all the images; you need to do the same. Note also the `getCachedPseudoStyle()` there which you should copy.